source/m2mtlvserializer.cpp

```diff
@@ -126,8 +126,21 @@ bool M2MTLVSerializer::serialize_resource(const M2MResource *resource, uint8_t *
 {
     bool success = false;
     if(resource->name_id() != -1) {
-        success = serialize_TILV(TYPE_RESOURCE, resource->name_id(),
+        if ( resource->resource_instance_type() == M2MResourceInstance::INTEGER) {
+            int64_t valueInt = resource->get_value_int();
+            uint32_t buf_size;
+            /* max len 8 bytes */
```

Comment:
Size is fixed, isn't it? Or is it supposed to be variable? Spec suggests there are 4 fixed-width integer types, this only does 64-bit.